ddtrace/llmobs/_evaluators/ragas/faithfulness.py

```diff
@@ -106,23 +122,31 @@ def __init__(self, llmobs_service):
     def run_and_submit_evaluation(self, span_event: dict):
         if not span_event:
             return
-        score_result = self.evaluate(span_event)
-        if score_result is not None:
+        score_result_or_failure = self.evaluate(span_event)
+        telemetry_writer.add_count_metric(
+            "llmobs",
+            "evaluators.{}.run".format(self.LABEL),
+            1,
+            tags=(("state", score_result_or_failure if isinstance(score_result_or_failure, str) else "success"),),
+        )
+        if isinstance(score_result_or_failure, float):
             self.llmobs_service.submit_evaluation(
                 span_context={"trace_id": span_event.get("trace_id"), "span_id": span_event.get("span_id")},
                 label=RagasFaithfulnessEvaluator.LABEL,
                 metric_type=RagasFaithfulnessEvaluator.METRIC_TYPE,
-                value=score_result,
+                value=score_result_or_failure,
             )
 
-    def evaluate(self, span_event: dict) -> Optional[float]:
+    def evaluate(self, span_event: dict) -> Union[float, str]:
```

Comment:
changed the signature here to return a failure reason so we can be granular in tracking ragas failures .. i especially care if there are a lot of `fail_extract_faithfulness_inputs` errors since that likely means users are annotating context information wrong. Would justify work on automatically inferring inputs